src/components/SampleView/SampleView.tsx

```diff
@@ -18,47 +22,100 @@ function formatSlotValue(value: SampleDataValue) {
+
+  const schemaClass = TEMPLATES[packageName].schemaClass;
+  const hiddenSlots = getHiddenSlotsForSchemaClass(schemaClass);
   const slotGroups = schemaClass ? groupClassSlots(schema, schemaClass) : [];
 
+  const handleDismiss = () => {
+    setHiddenSlotsForSchemaClass(schemaClass, []);
+  };
+
   if (!sample) {
     return null;
   }
 
   return (
     <>
+      {hiddenSlots === undefined && (
+        <IonList className={styles.slotSelectorBanner}>
+          <IonItem lines="none">
+            <IonLabel>Too many fields?</IonLabel>
+            <IonButton
+              slot="end"
+              fill="clear"
+              onClick={() => setIsModalOpen(true)}
+            >
+              Customize List
+            </IonButton>
+            <IonButton slot="end" fill="clear" onClick={handleDismiss}>
+              Dismiss
+            </IonButton>
+          </IonItem>
+        </IonList>
+      )}
+
       {slotGroups.map((group) => (
         <React.Fragment key={group.name}>
           <SectionHeader>{group.title}</SectionHeader>
-          {group.slots.map((slot) => (
-            <IonItem key={slot.name} onClick={() => onSlotClick(slot)}>
-              {validationResults?.[slot.name] && (
-                <IonIcon
-                  aria-hidden="true"
-                  icon={warningOutline}
-                  color="warning"
-                  slot="end"
-                />
-              )}
-              <IonLabel>
-                <h3>{slot.title || slot.name}</h3>
-                <p>{formatSlotValue(sample?.[slot.name])}</p>
-              </IonLabel>
-            </IonItem>
-          ))}
+          <IonList className="ion-padding-bottom">
+            {group.slots.map(
+              (slot) =>
+                (hiddenSlots === undefined ||
+                  !hiddenSlots.includes(slot.name)) && (
+                  <IonItem key={slot.name} onClick={() => onSlotClick(slot)}>
+                    {validationResults?.[slot.name] && (
+                      <IonIcon
+                        aria-hidden="true"
+                        icon={warningOutline}
+                        color="warning"
+                        slot="end"
+                      />
+                    )}
+                    <IonLabel>
+                      <h3>{slot.title || slot.name}</h3>
+                      <p>{formatSlotValue(sample?.[slot.name])}</p>
+                    </IonLabel>
+                  </IonItem>
+                ),
+            )}
+          </IonList>
         </React.Fragment>
       ))}
+
+      {hiddenSlots !== undefined && hiddenSlots.length > 0 && (
+        <IonList className="ion-padding-bottom">
+          <IonItem lines="none" onClick={() => setIsModalOpen(true)}>
+            <IonLabel class="ion-text-wrap">
+              <p>
+                Not seeing a field you were looking for? Tap here to update
+                field visibility settings.
+              </p>
+            </IonLabel>
+          </IonItem>
```

Comment:
I am wondering whether any accessibility issues (e.g. inability to do "keyboard navigation" on the PWA, or some iOS accessibility helper not treating it like an interactive element) will come up due to this `IonItem` not having an `href` or `button` property set. I see a section in the Ionic docs about "Clickable Items": https://ionicframework.com/docs/api/item#clickable-items

Reply:
Good thought. I will take a closer look at that.

Reply:
I added the `button` and `detail={false}` props to `IonItem` as suggested by the docs.

Reply:
Cool—thank you!